Fix part of #8472: Migrate editable-story-backend-api.service

darksun27 opened the pull request:
## Overview

1. This PR fixes or fixes part of #8472 .
2. This PR does the following: This PR migrates editable-story-backend-api.service.ts to Angular 8.
## Essential Checklist

- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The linter/Karma presubmit checks have passed locally on your machine.
- [x] "Allow edits from maintainers" is checked. (See [here](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork) for instructions on how to enable it.)
  - This lets reviewers restart your CircleCI tests for you.
- [x] The PR is made from a branch that's **not** called "develop".

darksun27 committed:
Migrate Editable Story Backend API Service

darksun27 committed:
Merge remote-tracking branch 'upstream/develop' into migrate-editable-story-backend

darksun27 commented:
@srijanreddy98 PTAL!

srijanreddy98 commented on core/templates/domain/story/editable-story-backend-api.service.ts:
```diff
@@ -16,32 +16,67 @@
+      'version': number,
+      'story_contents': {
+        'initial_node_id': string,
+        'nodes': [{
```
This is an array of type StoryNodeBackendDict?

darksun27 replied:
The array is of the type StoryNodeBackendDict but the StoryContentBackendDict contains it which is contained by the StoryBackendDict.

Changed!

srijanreddy98 commented on core/templates/domain/story/editable-story-backend-api.service.ts:
```diff
@@ -16,32 +16,67 @@
+import { StoryDomainConstants } from 'domain/story/story-domain.constants';
+import { UrlInterpolationService } from 'domain/utilities/url-interpolation.service';
+
+export interface PublishedStoryModel {
```
The names for interfaces, whose keys contain "_", should be suffixed with "BackendDict". This should be PublishedStoryBackendDict.

darksun27 replied:
Renamed!

srijanreddy98 commented on core/templates/domain/story/editable-story-backend-api.service.ts:
```diff
@@ -16,32 +16,67 @@
+    }],
+    'topic_url_fragment': string,
+    'classroom_url_fragment': string
+  }
```
The type exists in StoryNodeObjectFactory. It is called StoryBackendDict.

darksun27 replied:
Yup used it but it was a subset of this object.

darksun27 committed:
Merge remote-tracking branch 'upstream/develop' into migrate-editable-story-backend

srijanreddy98 reviewed (changes requested):
@darksun27 took a pass. PTAL!

darksun27 committed:
Fix Nits: Refactor the code
- Remove the if conditions for callbacks
- Renamed the PublishedStroyModel to BackendDict
- Included the StoryObjectFactory into the BackendDict

darksun27 committed:
Merge branch 'migrate-editable-story-backend' of
https://github.com/darksun27/oppia into migrate-editable-story-backend
Changed remotely, so pulling the code.

darksun27 committed:
Merge remote-tracking branch 'upstream/develop'
into migrate-editable-story-backend

darksun27 commented:
@srijanreddy98 PTAL!

srijanreddy98 reviewed (approved):
LGTM!

srijanreddy98 commented:
@nithusha21 @Showtim3 PTAL!

darksun27 commented:
Helping Reshu Kumari Migrate this service! Closing this PR.
